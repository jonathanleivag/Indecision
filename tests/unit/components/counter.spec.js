import Counter from '@/components/Counter'
import { shallowMount } from '@vue/test-utils'

describe('Counter Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Counter)
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render in h2 tag the default value', () => {
    const h2 = wrapper.find('h2')
    expect(h2.text()).toBe('Contador')
  })
  it('increase counter', async () => {
    const increaseBtn = wrapper.find('button')
    await increaseBtn.trigger('click')
    const p = wrapper.find('[data-test-id="counter"]')
    expect(p.text()).toBe('101')
  })

  it('decrease counter', async () => {
    const increaseBtn = wrapper.findAll('button')
    const btn = increaseBtn[1]
    await btn.trigger('click')
    await btn.trigger('click')

    const p = wrapper.find('[data-test-id="counter"]')
    expect(p.text()).toBe('98')
  })

  it('props default', async () => {
    const wrapper = shallowMount(Counter)
    const { start } = wrapper.props()
    expect(typeof start).toBe('number')
    const p = wrapper.find('[data-test-id="counter"]')
    expect(+p.text()).toBe(start)
  })

  it('change props title', async () => {
    const title = 'Hola mundo'
    const wrapper = shallowMount(Counter, { props: { title } })
    const h2 = wrapper.find('h2').text()
    expect(h2).toBe(title)
  })
})
