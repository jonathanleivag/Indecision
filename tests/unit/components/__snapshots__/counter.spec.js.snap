// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should match the snapshot 1`] = `
<h2>Contador</h2>
<p>100 <sup>2</sup> = 10000</p>
<p data-test-id="counter">100</p>
<div><button>+1</button><button>-1</button></div>
`;
